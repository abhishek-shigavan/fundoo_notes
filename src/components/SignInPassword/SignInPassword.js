import React, { Component } from "react";
import { Link } from "react-router-dom";
import { DataRequest } from "../../services/RequestService";
import axios from "axios";
import { Box, TextField, InputAdornment, IconButton, Button, Grid, Typography, Chip } from "@material-ui/core";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import AccountCircleIcon from "@mui/icons-material/AccountCircle";
import { styled } from "@mui/material/styles";
import { blue } from "@mui/material/colors";

class SignInPassword extends Component {
constructor() {
        super();
        this.state = {
            emailId: "",
            password: "",
            showPassword: false,
            passwordError: ""
        }
    }

    handleClickShowPassword() {
        const Password = this.state.showPassword;
        this.setState({showPassword: !Password});
    };

    handleMouseDownPassword(event) {
        event.preventDefault();
    };

    validatePassword(event) {
        const currentPassword = event.target.value;
        if( currentPassword.length > 0 && currentPassword.length < 6) {
            this.setState({password : "", passwordError: "Password length should be 6"});
        }
        else{
            this.setState({password : event.target.value, passwordError:"",
                            emailId : this.props.email});
        }       
    }

    handleSubmit = (event) => {
        event.preventDefault();
        const { emailId } = this.state;
        // console.log(emailId);
        const { password } = this.state;
        // const User = {
        //     email: emailId,
        //     password: password
        // };

        // const response = await axios.post(`${process.env.REACT_APP_BASE_URL}/user/login`, User,);
        // console.log(response);
        // console.log(response.status);
        // this.props.history.push(`/Dashboard`);    
        // alert('In get User');
        const User = {
            email: emailId,
            password: password
        };

        this.props.sendLoginData(User);

        // DataRequest(User, "/user/login").then((res)=>{
        //     if(res.status == 200) {
        //         this.props.history.push("/dashboard")
        //     }
        // })

    }


    ColorButton = styled(Button)(({ theme }) => ({
        color: theme.palette.getContrastText(blue[500]),
        backgroundColor: blue[500],
        width: '120px',
        height: '36px',
        fontSize: '1.3rem',
        '&:hover': {
            backgroundColor: blue[700],
        },
    }));

    WOBackgroundButton = styled(Button)(({ theme }) => ({
        color: blue[600],
        width: '140px',
        height: '36px',
        fontSize: '1.3rem',
        padding: 0,

        '&:hover': {
            backgroundColor: blue[50],
        },
    }));

    render() {
        return(
            <Box data-testid="sign-in-password"
            sx={{
            display: 'flex',
            width: '100%',
            height: '100vh',
            alignItems: 'center',
            justifyContent: 'center'
            }}>
                <Box component="form" onSubmit={(e)=>{this.handleSubmit(e)}} sx={{
                fontSize: '62.5%',
                width: '27vw',
                height: '63.3vh',
                paddingTop: '4.8rem',
                paddingBottom: '3.6rem',
                paddingLeft: '4rem',
                paddingRight: '4rem',
                border: '1px solid #dadce0', 
                borderRadius: '0.8rem',
                alignSelf: 'center',
                }}>
                    <Grid container spacing={7} justifyContent="center">
                        <Grid item lg={12}>
                            <Typography variant="h4" align="center" color="error">
                                FUNDO
                            </Typography>
                        </Grid>
                        <Grid item lg={12} style={{paddingRight:'105px', paddingLeft:'105px', paddingTop:0, paddingBottom:0}}>
                            <Typography variant="h4" align="center">
                                Welcome
                            </Typography><br />
                            <Chip
                                clickable 
                                icon={<AccountCircleIcon sx={{fontSize: '2rem'}} />} 
                                label={this.props.email} 
                                variant="outlined" 
                                style={{ width: '214px', height:'35px',
                                fontSize: '1.3rem'}}
                                />
                        </Grid>
                        <Grid item lg={12}>
                            <TextField
                                id="user-password"
                                label="Enter Your Password"
                                variant="outlined"
                                size="medium"
                                type="text"
                                helperText={this.state.passwordError}
                                fullWidth
                                required
                                style={{fontSize:'2rem'}}
                                type={this.state.showPassword ? "text" : "password"}
                                onChange={(event) => { this.validatePassword(event) }}
                                InputProps={{
                                    endAdornment: (
                                        <InputAdornment position="end">
                                            <IconButton
                                                aria-label="toggle password visibility"
                                                onClick={() => this.handleClickShowPassword()}
                                                onMouseDown={(event) => this.handleMouseDownPassword(event)}
                                                edge="end">
                                                {this.state.showPassword ? (<VisibilityOff />) : (<Visibility />)}
                                            </IconButton>
                                        </InputAdornment>
                                    ),
                                }}
                            />
                        </Grid>
                        <Grid item lg={12} container justifyContent="space-between">
                            <Grid item md={4} sm={12}>
                                <this.WOBackgroundButton onClick={()=>{<Link to="/forgotpassword" />}}>Forgot Password?</this.WOBackgroundButton>
                            </Grid>
                            <Grid item md={4} sm={12}>
                                <this.ColorButton type="submit">Login</this.ColorButton>
                            </Grid>
                        </Grid>
                    </Grid>
                </Box>
            </Box>
        );
    }
}

export default SignInPassword;