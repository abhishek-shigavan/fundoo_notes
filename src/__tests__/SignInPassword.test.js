import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import SignInPassword from '../components/SignInPassword/SignInPassword';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it("component renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SignInPassword></SignInPassword>, div)
});

it("renders SignInPassword with props value correctly", () => {
    const { getByTestId } = render(<SignInPassword email="peterparkar@gmail.com"></SignInPassword>)
    expect(getByTestId('sign-in-password')).toHaveTextContent("peterparkar@gmail.com");
});

it("matches snapshot", ()=> {
    const virtualDom = renderer.create(<SignInPassword label="something@todo.com"></SignInPassword>).toJSON();
    expect(virtualDom).toMatchSnapshot();
})
