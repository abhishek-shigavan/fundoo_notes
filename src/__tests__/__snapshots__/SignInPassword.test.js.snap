// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-9"
  data-testid="sign-in-password"
>
  <form
    className="MuiBox-root MuiBox-root-10"
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-7 MuiGrid-justify-content-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-colorError MuiTypography-alignCenter"
        >
          FUNDO
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
        style={
          Object {
            "paddingBottom": 0,
            "paddingLeft": "105px",
            "paddingRight": "105px",
            "paddingTop": 0,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
        >
          Welcome
        </h4>
        <br />
        <div
          className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-clickable"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          style={
            Object {
              "fontSize": "1.3rem",
              "height": "35px",
              "width": "214px",
            }
          }
          tabIndex={0}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon css-1q13jnw-MuiSvgIcon-root"
            data-testid="AccountCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
          <span
            className="MuiChip-label"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style={
            Object {
              "fontSize": "2rem",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="user-password"
            id="user-password-label"
          >
            Enter Your Password
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              disabled={false}
              id="user-password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="toggle password visibility"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-5"
              >
                <span>
                  Enter Your Password
                   *
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-space-between MuiGrid-grid-lg-12"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text css-15ea7oo"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Forgot Password?
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text css-1qox269"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Login
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
